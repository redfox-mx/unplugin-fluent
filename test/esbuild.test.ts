import { resolve } from 'node:path'
import { describe, expect, it } from 'vitest'
import { build } from 'esbuild'
import fluent from '../src/esbuild'

async function getCode() {
  const out = (await build({
    entryPoints: [resolve(__dirname, 'fixture/importer.js')],
    bundle: true,
    format: 'esm',
    plugins: [fluent()],
    write: false,
  }))

  return out.outputFiles.map(file => file.text).join('\n')
}

describe('esbuild', () => {
  describe('fixtures', () => {
    it('compiles fluent files', async () => {
      const code = await getCode()
      expect(code).toMatchSnapshot()
    })
  })
})
