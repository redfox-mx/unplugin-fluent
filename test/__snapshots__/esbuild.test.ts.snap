// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`esbuild > fixtures > compiles fluent files 1`] = `
"// test/fixture/locale.ftl
var locale_default = { body: [{ "id": "hello-user", "value": ["Hello, ", { "type": "var", "name": "userName" }, "!"], "attributes": {} }, { "id": "shared-photos", "value": [{ "type": "var", "name": "userName" }, " ", { "type": "select", "selector": { "type": "var", "name": "photoCount" }, "variants": [{ "key": { "type": "str", "value": "one" }, "value": "added a new photo" }, { "key": { "type": "str", "value": "other" }, "value": ["added ", { "type": "var", "name": "photoCount" }, " new photos"] }], "star": 1 }, " to ", { "type": "select", "selector": { "type": "var", "name": "userGender" }, "variants": [{ "key": { "type": "str", "value": "male" }, "value": "his stream" }, { "key": { "type": "str", "value": "female" }, "value": "her stream" }, { "key": { "type": "str", "value": "other" }, "value": "their stream" }], "star": 2 }, "."], "attributes": {} }] };

// test/fixture/importer.js
var output = locale_default;
export {
  output
};
"
`;
